src/PostgREST/DbStructure.hs

```diff
@@ -699,18 +699,11 @@ pkFromRow tabs (s, t, n) = PrimaryKey <$> table <*> pure n
   where table = find (\tbl -> tableSchema tbl == s && tableName tbl == t) tabs
 
 -- returns all the primary and foreign key columns which are referenced in views
-pfkSourceColumns :: [Column] -> PgVersion -> Bool -> H.Statement ([Schema], [Schema]) [SourceColumn]
-pfkSourceColumns cols pgVer =
+pfkSourceColumns :: [Column] -> Bool -> H.Statement ([Schema], [Schema]) [SourceColumn]
+pfkSourceColumns cols =
   H.Statement sql (contrazip2 (arrayParam HE.text) (arrayParam HE.text)) (decodeSourceColumns cols)
-  -- query explanation at https://gist.github.com/steve-chavez/7ee0e6590cddafb532e5f00c46275569
```

Comment:
@wolfgangwalther I think this one will be hard to understand by just looking at the code. Even on my prior query I had to think hard to remember what I did.

Could you add a comment here explaining the general process?

> First the pg_node_tree is reformatted to proper json with a series of replace() and regexp_replace() calls. After that, a cast to json is possible and the TargetEntries can easily be extracted.

Even a comment like the above would be helpful.

If you have some sample node trees like on my prior gist, that would be helpful as well. You can also leave a gist of your own if you'd like to.

Reply:
Yes that makes a lot of sense. I removed the link to your gist, because big parts of it don't apply anymore (the regex stuff). But the basics still do and you're absolutely right that this could use a bit more explanation here. I tried to comment the json conversion as much as possible - but that does not give any idea of the bigger picture here.

I will try to add more comments.

Reply:
I forked your gist and took the regex stuff out. This is part #1. Then I wrote part #2 about the json transformation in detail.

First of all, this is lots of detail, very good to come back to that in the future. But even better - explaining it in detail, I realized that I could skip one of the replaces altogether. My timings are now:

| query | execution |
| -- | --: |
| old master | 406.5 |
| master | 229.8 |
| json before | 317.5 |
| json now | 301.8 |

Reply:
You probably won't notice on your super fast machine, but on my slow Ryzen it's 16ms faster ;)

Reply:
Awesome! That's a really nice explanation Wolfgang, I feel we have at least 2 blog posts with those gists :)

> You probably won't notice on your super fast machine, but on my slow Ryzen it's 16ms faster ;)

I'll try the new query to see if I can beat your time again :smile:. Do you have a quick pastebin for me? :)

Reply:
https://pastebin.com/6x3wTJVX

Here you go :)

Reply:
| query       | execution          |
| ----------- | ------------------ |
| old master  | 282.7              |
| master      | 168.6              |
| json before | 249.2              |
| json now    | 232.0              |

17 ms faster for me. Beat you again :laughing:

Reply:
Ah, wait a second. Between my latest test and yours... I removed another replace. It didn't show much of a difference in perf, so I didn't post it... but your test was running with both of them removed..

So let me run my tests again... let's see whether I can get another 1 or 2 ms. :D

Reply:
319.9-303.5 = 16.4. :(

I'm leaving now to do something else. This is annoying :D